### ISSUE TITLE
Improve RepositoryTrack, ContractDetails, DetailsDataStore

### ISSUE THREAD

**Issue by aeidelman** [rsksmart/rskj#11]
Current implementation is a bit convoluted (see updateBatch implementation in RepositoryImpl and RepositoryTrack). The cache of ContractDetails kept by DetailsDataStore is SHARED by all Repository objects. The manage of originalContractDetails in updateBatch repository methods is not clear and direct. DetailsDataStore arbitrary close all the contract stores in its cache, when flush

**Comment by collheesi** [comment]
I think this is outdated now, especially after #832. We should open a new issue for the new Repository efforts that we're planning.
